x-pack/test/security_functional/fixtures/common/test_endpoints/server/init_routes.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { CoreSetup } from '../../../../../../../src/core/server';
+
+export function initRoutes(core: CoreSetup) {
+  const authenticationAppOptions = { simulateUnauthorized: false };
+  core.http.resources.register(
+    {
+      path: '/authentication/app',
+      validate: false,
+    },
+    async (context, request, response) => {
+      if (authenticationAppOptions.simulateUnauthorized) {
```

Comment:
__note:__ the idea here is that we're trying to simulate the case when authentication pipeline managed to authenticate request, but the handler itself threw `401` for some reason anyway, e.g. improper handling of errors originated from using the custom ES credentials (we had similar SDH in the past).

Reply:
I think this is an elegant way to handle this functional test!

Is there a specific reason you didn't also include a new test for `oidc/url_capture.ts`?

Reply:
> I think this is an elegant way to handle this functional test!

Thanks!

> Is there a specific reason you didn't also include a new test for oidc/url_capture.ts?

Yeah, I guess inattentiveness can be an excuse, but not here! Will add test for ` oidc/url_capture.ts` as well, thanks!